ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/blockhash/CancunBlockHashProcessor.java

```diff
@@ -14,28 +14,46 @@
  */
 package org.hyperledger.besu.ethereum.mainnet.blockhash;
 
+import org.hyperledger.besu.datatypes.Address;
 import org.hyperledger.besu.ethereum.core.ProcessableBlockHeader;
-import org.hyperledger.besu.ethereum.mainnet.ParentBeaconBlockRootHelper;
 import org.hyperledger.besu.ethereum.mainnet.systemcall.BlockProcessingContext;
-import org.hyperledger.besu.evm.worldstate.WorldUpdater;
+import org.hyperledger.besu.ethereum.mainnet.systemcall.InvalidSystemCallAddressException;
+import org.hyperledger.besu.ethereum.mainnet.systemcall.SystemCallProcessor;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes32;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Processes the beacon block storage if it is present in the block header. */
 public class CancunBlockHashProcessor extends FrontierBlockHashProcessor {
+  private static final Logger LOG = LoggerFactory.getLogger(CancunBlockHashProcessor.class);
+  Address BEACON_ROOTS_ADDRESS =
+      Address.fromHexString("0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02");
```

Comment:
Is this class always used when there is a cancun fork? We might have to make this configurable, because e.g. an L2 might want to support cancun, but cannot create a contract at that address.

Reply:
If there is no contract or no code in the contract, the system call fails silently.

Reply:
I can't see how this PR changed this, it isn't the same thing before?